saul-core/src/main/java/edu/illinois/cs/cogcomp/saul/learn/MultilayerPerceptron.java

```diff
@@ -0,0 +1,588 @@
+     * @param num_inputs Number of dimensions for (input) feature vector
+     * @param num_outputs Number of nodes at output layer. For classification tasks, use one output node for each
+     * possible class label.
+     * @param hidden_layers Specify hidden layers. For example, {5,6,5} describes a neural network with three
+     * fully connected hidden layers having sizes of 5,6,5 respectively.
+     * @param activation_function Specify the activation functions for the layers, based on the codes for each function.
+     * Number of entries in this array must be equal to 1 plus the number of hidden layers. The last entry corresponds
+     * to activation of the output layer, which should always be sigmoid. For example, a network
+     * with one hidden layer with a tanh at hidden layer and sigmoid at output layer can be specified with {2,1}.
+     */
+    public MultilayerPerceptron(int num_inputs, int num_outputs, int[] hidden_layers, int [] activation_function){
+        super("");
+        int[] layer_size = new int[hidden_layers.length+2];
+        layer_size[0] = num_inputs;
+        for (int i = 0; i < hidden_layers.length; i++){
+            layer_size[i+1] = hidden_layers[i];
+        }
+        layer_size[hidden_layers.length+1] = num_outputs;
+
+        layers = new Layer[hidden_layers.length+1];
+        for (int i = 0; i < layers.length; i++){
+            layers[i] = new Layer(layer_size[i],layer_size[i+1],activation_function[i]);
+        }
+    }
+
+    /**
+    * Trains the neural network on a given example, given a real input and output vector, using default learning parameters.
+    *
+    * @param input      Real vector consisting of the input features of an example
+    * @param output     Real vector consisting of the labelled outputs of the neural network.
+    */
+    public void learn(double[] input, double[] output){
+        learn(input,output,default_learning_rate,default_lambda,default_momentum);
+    }
+
+    /**
+     * Trains the neural network on a given example, given a real input and output vector, using given learning parameters.
+     *
+     * @param input         Real vector consisting of the input features of an example
+     * @param output        Real vector consisting of the labelled outputs of the neural network.
+     * @param learning_rate Learning rate
+     * @param lambda        Penalization constant for L-2 regularization
+     * @param momentum      Momentum value for speeding convergence
+     */
+    public void learn(double[] input, double[] output, double learning_rate, double lambda, double momentum){
+        reset_errors();
+        compute_activation(input);
+        output_errors(output);
+        compute_gradients(input);
+        update_weights(learning_rate,lambda, momentum);
+    }
+
+    /**
+     * Trains the neural network on a sparse input vector, using default learning parameters.
+     *
+     * @TODO Only works for binary classification now.
```

Comment:
could you clarify this? Do you mean, the output vector can only be a vector of 1.0/0.0 ?